src/app/market/sell/add-item/add-item.component.ts

```diff
@@ -300,13 +302,13 @@ export class AddItemComponent implements OnInit, OnDestroy {
     this.log.d('Saving as a template.');
     if (this.templateId) {
       // update
-      this.update();
+      this.itemFormGroup.valid ? this.update() : this.snackbar.open('Invalid Listing');
```

Comment:
I don't think this check is still required?
due to
```
+    if (this.itemFormGroup.invalid) {
+      return Observable.throw('Invalid Listing');
+    }
```

Reply:
This check is for updating existing template of not-yet-published listing, it calls `update` method *only* when form is valid.
Without this check user able to update existing template with invalid values.
Also at this point `update` itself need to be fixed - and this check could be moved into `update` method.
What do you think?